### DISCUSSION TITLE
CORS ERROR

### DISCUSSION THREAD

**Discussion by 4august** [vercel/community#5972] [Help]
### Summary

Im trying to use my api alredy published in vercel but everytime I try to make login the cors blocks my requistion. The error is **"Access to XMLHttpRequest at 'https://financas-backend-one.vercel.app/login' from origin 'https://cliente-financas.vercel.app' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource."** I alredy tried some solutions in stackoverflow like https://stackoverflow.com/questions/70099946/how-to-enable-cors-not-working-on-vercel and still not working. I really need HELP to resolve this. 

### Steps to Reproduce

My first version o vercel.json is `{
    "version": 2,
    "builds":[
        {
            "src": "src/main.ts",
            "use": "@vercel/node"
        }
    ],
    "routes":[
        {
            "src": "/(.*)",
            "dest": "src/main.ts",
            "methods": [
                "GET",
                "POST",
                "PUT",
                "PATCH",
                "DELETE"
            ]
        }
    ]
}` I tried the solutions from https://stackoverflow.com/questions/70099946/how-to-enable-cors-not-working-on-vercel but no success

**Comment by samkim28** [comment]
I struggled through the same problem and looked at almost every combination of `vercel.json` headers. For me, the answer was turning deployment protection off in the project settings. I know I'll eventually turn it back on for certain environments but this was blocking me on dev! Hope that helps.

**Reply by 4august** [reply]
removing protection still mantain the ssl certificate?????
